lib/utils/docs.go

```diff
@@ -0,0 +1,299 @@
+		}
+	}
+	return flagString
+}
+
+// cmdModelCollection is used to sort CmdModels for display in the CLI reference
+// docs page.
+type cmdModelCollection []*kingpin.CmdModel
+
+// Len returns the length of c. Required to implement sort.Interface.
+func (c cmdModelCollection) Len() int {
+	return len(c)
+}
+
+// Less returns whether command name i is lexicographically ordered before
+// command name j. Required to implement sort.Interface.
+func (c cmdModelCollection) Less(i, j int) bool {
+	return c[i].FullCommand < c[j].FullCommand
+}
+
+// Swap reverses the ordering of i and j within c. Required to implement
+// sort.Interface.
+func (c cmdModelCollection) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+
+// sortcommandsByName sorts the commands in cmds by their full command names,
+// including all subcommands.
+func sortCommandsByName(cmds []*kingpin.CmdModel) []*kingpin.CmdModel {
+	col := cmdModelCollection(cmds)
+	sort.Stable(&col)
+	return []*kingpin.CmdModel(col)
+}
+
+// formatDefaultFlagValue returns the default value of flag to display in a
+// table of flags. Assumes that a Boolean flag is false unless it is true by
+// default.
+func formatDefaultFlagValue(flag *kingpin.FlagModel) string {
+	switch {
+	case len(flag.Default) == 0 && flag.IsBoolFlag():
+		return "`false`"
+	case len(flag.Default) > 0:
+		ret := make([]string, len(flag.Default))
+		for i, v := range flag.Default {
+			ret[i] = fmt.Sprintf("`%v`", v)
+		}
+		return strings.Join(ret, ",")
+	default:
+		return "none"
+	}
+}
+
+// formatDefaultArgValue returns the default value of arg to display in a table
+// of flags. It also indicates whether the value is optional or required.
+func formatDefaultArgValue(arg *kingpin.ArgModel) string {
+	var ret string
+	if len(arg.Default) > 0 {
+		def := make([]string, len(arg.Default))
+		for i, v := range arg.Default {
+			def[i] = fmt.Sprintf("`%v`", v)
+		}
+		ret = strings.Join(def, ",")
+	} else {
+		ret = "none"
+	}
+	if arg.Required {
+		ret += " (required)"
+	} else {
+		ret += " (optional)"
+	}
+
+	return ret
+}
+
+// repeatableFlag is an interface for flags that can be repeated. Unexported
+// type in github.com/alecthomas/kingpin/v2.
+type repeatableFlag interface {
+	IsCumulative() bool
+}
+
+// formatUsageArg prints a command argument to include in a usage snippet.
+func formatUsageArg(arg *kingpin.ArgModel) string {
+	var ret string
+	switch {
+	case arg.PlaceHolder != "":
+		ret = arg.PlaceHolder
+	// Some special cases have empty arg names
+	case arg.Name == "":
+		ret = "args"
+	default:
+		ret = "<" + arg.Name + ">"
+	}
+	if v, ok := arg.Value.(repeatableFlag); ok && v.IsCumulative() {
+		ret += "..."
+	}
+	if !arg.Required {
+		ret = "[" + ret + "]"
+	}
+	return ret
+}
+
+// formatHelp prints help text to include in a Markdown table cell. It escapes
+// curly braces to avoid breaking the MDX parser.
+func formatHelp(help string) string {
+	return strings.NewReplacer("{", `\{`, "}", `\}`).Replace(help)
+}
```

Comment:
I think we'll need to escape pipes too.
```suggestion
// formatHelp prints help text to include in a Markdown table cell. It escapes
// curly braces to avoid breaking the MDX parser, and it escapes pipes to
// avoid breaking the cell.
func formatHelp(help string) string {
	return strings.NewReplacer("{", `\{`, "}", `\}`, "|", `\|`).Replace(help)
}
```

Reply:
Thanks!